lab-michael/routes/recipe-routes.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const storage = require('../lib/storage.js');
+const response = require('../lib/response.js');
+const Recipe = require('../model/recipe.js');
+
+module.exports = function(router) {
+  router.get('/api/recipe', function(req, res){
+    if(req.url.query.id){
+      storage.fetchItem('recipe', req.url.query.id)
+      .then(recipe => {
+        res.writeHead(200, {
+          'Content-Type': 'application/json',
+        });
+        res.write(JSON.stringify(recipe));
+        res.end();
+      })
+      .catch(err => {
+        console.error(err);
+        res.writeHead(404, {
+          'Content-Type': 'text/plain',
+        });
+        res.write('not found\n');
+        res.end();
+      });
+      return;
+    } else {
+      if(!req.url.query.id){
+        storage.fetchAll('recipe', req.url.query.id)
+        .then(recipe => {
+          res.writeHead(200, {
+            'Content-Type': 'application/json',
+          });
+          res.write(JSON.stringify(recipe));
+          res.end();
+        })
+        .catch(err => {
+          console.error(err);
+          res.writeHead(404, {
+            'Content-Type': 'text/plain',
+          });
+          res.write('not found\n');
+          res.end();
+        });
+        return;
+      }
+      res.writeHead(400, {
+        'Content-Type': 'text/plain',
+      });
+      res.write('bad request');
+      res.end();
+    // storage.fetchItem()
+    }
+  });
+
+
+  //should create a recipe on the storage
+  router.post('/api/recipe', function(req, res){
+    // console.log(req.body);
+    try {
+      var recipe = new Recipe(req.body.name, req.body.content, req.body.mealType);
```

Comment:
I would argue that the new instance of an object should be created within the `createItem` method rather than here at this point.